tests/ProxyManagerTestAsset/RemoteProxy/RemoteServiceWithDefaultsInterface.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ProxyManagerTestAsset\RemoteProxy;
+
+/**
+ * Simple interface for a remote API that methods with default parameters
+ *
+ * @author Pedro Tanaka <pedro.stanaka@gmail.com>
+ * @license MIT
+ */
+interface RemoteServiceWithDefaultsInterface
+{
+    public function optionalNonNullable(string $foo, string $optional = ''): int;
```

Comment:
Let's make this more explicit, by having `$optional = 'I am an optional parameter to be preserved during calls'` or such